src/convertSearchParams.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+function convertSearchParams(params) {
+  const query = {};
+
+  params.split('&').forEach(req => {
+    const key = decodeURIComponent(req.split('=')[0]);
+    const value = decodeURIComponent(req.split('=')[1]);
+
+    if (query.hasOwnProperty(key)) {
+      if (Array.isArray(query[key])) {
+        query[key].push(value);
+      } else {
+        const valueInArray = [query[key], value];
+
+        query[key] = valueInArray;
+      }
+    } else {
+      query[key] = value;
+    }
+  });
+
+  return query;
+}
```

Comment:
Can be much simpler

Reply:
const query = Object.fromEntries(normalizedUrl.searchParams.entries()) - this one? it retern only last value

give me please an example or a link where can I read how can I get an array of searchParams with one key

Reply:
`const currentUrlSearchParams = new URLSearchParams(params);
  const query = Object.fromEntries(currentUrlSearchParams.entries());

  for (const key in query) {
    const allValues = currentUrlSearchParams.getAll(key)

    if (allValues.length > 1) {
      query[key] = allValues;
    }
  }`
  
  this solution return an array in query like:
  '/kata/search/my-languages?q=&r%5B%5D=-8&r%5B%5D=-7&r%5B%5D=-6&r%5B%5D=-5&r%5B%5D=-4&order_by=sort_date%20desc'
  
  does the solution const query = Object.fromEntries(normalizedUrl.searchParams.entries()) return an array to key 'r[]'?